File: blockchain/transaction.py
class Transaction:
    def __init__(self, sender, receiver, amount):
        self.sender = sender
        self.receiver = receiver
        self.amount = amount

    def to_string(self):
        return "Sender: " + self.sender + ", Receiver: " + self.receiver + ", Amount: " + str(self.amount)

File: blockchain_integration/pi_network/pi-browser/transaction.js
import { hash } from 'hash-sum';
import { elliptic } from 'elliptic';

class Transaction {
  constructor(sender, recipient, amount) {
    this.sender = sender;
    this.recipient = recipient;
    this.amount = amount;
    this.timestamp = Date.now();
    this.hash = hash(this.toString());
  }

  toString() {
    return `${this.sender}:${this.recipient}:${this.amount}:${this.timestamp}`;
  }

  sign(privateKey) {
    const signature = elliptic.ec('secp256k1').sign(this.hash, privateKey);
    return signature;
  }
}

export default Transaction;
